lib/headlines/security_headers/csp_directive.rb

```diff
@@ -0,0 +1,114 @@
+module Headlines
+  module SecurityHeaders
+    class CspDirective
+      SRC_DIRECTIVES = %w(child-src
+                          connect-src
+                          default-src
+                          font-src
+                          frame-src
+                          img-src
+                          media-src
+                          object-src
+                          script-src
+                          style-src)
+
+      attr_accessor :in_meta
+      attr_writer :sources
+
+      def initialize(directive, in_meta = false)
+        @directive = directive
+        @in_meta = in_meta
+      end
+
+      def name
+        @name ||= @directive.split(" ")[0]
+      end
+
+      def value
+        @value ||= sources.join(" ")
+      end
+
+      def sources
+        @sources ||= @directive.split(" ")[1..-1]
+      end
+
+      def sources_hosts
+        sources.map { |source| source.gsub(%r{(https?://)?(www.)?}, "") } - ["'none'", "'self'", "'*'"]
+      end
+
+      def invalid?
+        SRC_DIRECTIVES.include?(name) && (sources.include?("'*'") || sources.include?("'none'")) && sources.size > 1
+      end
+
+      def valid?
+        !invalid?
+      end
+
+      def allows_unsecured_http?
+        sources.include?("http:") || (http_domain_name? && !sources.include?("https:"))
+      end
+
+      def restrictive_default_settings?
+        name == "default-src" && value =~ /^('none'|'self')$/
```

Comment:
^ and $ are implicitly multiline, so this would still match "'none'\nsomethingElse"

It may not have much of an affect, but intent is more clear if \A and \z are used instead for all of these regexes